src/Compilers/Core/Portable/StrongName/StrongNameFileSystem.cs

```diff
@@ -1,63 +1,45 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.IO;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis
 {
+    /// <summary>
+    /// This is an abstraction over the file system which allows for us to do more thorough unit testing.
+    /// </summary>
     internal class StrongNameFileSystem
     {
         internal readonly static StrongNameFileSystem Instance = new StrongNameFileSystem();
-        protected StrongNameFileSystem()
+        internal readonly Func<string> _getTempPath;
+
+        internal StrongNameFileSystem(string tempPath = null)
         {
+            _getTempPath = tempPath == null
```

Comment:
>tempPath [](start = 27, length = 8)

Why not just save this to a field and have `GetTempPath` do this at invocation time?

Reply:
Because `Path.GetTempPath` does not have a constant return value. It reads several environment variables to calculate the result and those variables can change during process execution (and yes there are real build systems even at Microsoft that do this). 

---
In reply to: [245385750](https://github.com/dotnet/roslyn/pull/31986#discussion_r245385750) [](ancestors = 245385750)